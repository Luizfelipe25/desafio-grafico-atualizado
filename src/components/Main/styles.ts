import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #fff;
  color: #06092b;
  width: 100%;
  height: 100%;
`

export const Title = styled.h1`
  font-size: var(--medium);
  margin-bottom: var(--medium);
  margin-top: var(--medium);
  text-align: center;
`

export const Description = styled.h2`
  margin-bottom: 50px;
  align-items: center;
  text-align: center;
  justify-content: center;

  select {
    -webkit-appearance: none;
    -moz-appearance: none;
    -ms-appearance: none;
    appearance: none;
    outline: 0;
    box-shadow: none;
    border: 0 !important;
    background: #eeeeee;
    background-image: none;
    flex: 1;
    padding: 0 0.5em;
    color: #06092b;
    cursor: pointer;
    font-size: 1em;
    font-family: 'Open Sans', sans-serif;
  }

  select::-ms-expand {
    display: none;
  }
`

export const Chart = styled.section`
  text-align: center;
  align-items: center;
  max-height: 50%;
`
