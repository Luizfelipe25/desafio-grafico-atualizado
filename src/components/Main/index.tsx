import ChartConfigure from 'components/Chart'
import * as S from './styles'
const Main = () => (
  <S.Wrapper>
    <S.Title>React Avançado</S.Title>
    <S.Description>boilerplate</S.Description>
    <select id="filter">
      <option selected value="AllValues">
        Desde o Início
      </option>
      <option value="LastM">Ultimo Mês</option>
      <option value="Last3M">Ultimos 3 meses</option>
      <option value="LastY">Ultimo ano</option>
      <option value="Last2Y">Ultimos 2 anos</option>
    </select>
    <ChartConfigure />
  </S.Wrapper>
)

export default Main
