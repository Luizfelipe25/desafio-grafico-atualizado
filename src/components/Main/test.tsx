import { render, screen } from '@testing-library/react'

import Main from '.'

describe('testando a main', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avan√ßado/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
