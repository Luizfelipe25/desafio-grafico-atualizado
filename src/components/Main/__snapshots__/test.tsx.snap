// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testando a main should render the heading 1`] = `
<main>
  <h1>
    React Avan√ßado
  </h1>
</main>
`;
